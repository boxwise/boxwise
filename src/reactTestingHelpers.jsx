import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent } from "react-testing-library";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import thunk from "redux-thunk";
import { createStore, applyMiddleware } from "redux";

export { fireEvent, wait } from "react-testing-library";

// following suggestions in https://testing-library.com/docs/react-testing-library/setup#custom-render

export const typeText = (element, value) => {
  fireEvent.change(element, { target: { value } });
};

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
});

export function customRender(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {},
    reducer = state => state
  } = {}
) {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    </MuiThemeProvider>
  );
}

export { customRender as render };
