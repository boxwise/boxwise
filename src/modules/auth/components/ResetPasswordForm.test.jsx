import React from "react";

import { render, fireEvent, typeText } from "reactTestingHelpers";

import ResetPassword from "./ResetPasswordForm";

describe("ResetPasswordForm", () => {
  let container;
  let getByTestId;

  const handler = jest.fn(() => Promise.resolve());
  const clickSubmit = () => fireEvent.click(getByTestId("resetPasswordButton"));
  const enterEmailAddress = text => typeText(getByTestId("email"), text);

  beforeEach(() => {
    ({ container, getByTestId } = render(
      <ResetPassword resetPassword={handler} />
    ));
  });

  it("does not trigger resetPassword when email is not provided", () => {
    enterEmailAddress("xyz");
    expect(getByTestId("email").value).toEqual("xyz");
    expect(container).toMatchSnapshot();
    expect(handler).not.toBeCalled();
  });

  it("triggers resetPassword when email is provided", async () => {
    enterEmailAddress("test@example.com");
    clickSubmit();

    expect(container).toMatchSnapshot();
    expect(handler).toBeCalledWith({
      email: "test@example.com"
    });
  });

  it("renders correctly", () => {
    expect(container).toMatchSnapshot();
  });
});
