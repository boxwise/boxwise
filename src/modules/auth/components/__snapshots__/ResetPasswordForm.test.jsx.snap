// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm does not trigger resetPassword when email is not provided 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    >
      <label
        class="MuiFormLabel-root-16 MuiFormLabel-error-19 MuiInputLabel-error-8 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
        data-shrink="false"
        for="email"
      >
        Email
      </label>
      <div
        class="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-error-42 MuiInput-error-28 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
      >
        <input
          aria-describedby="email-helper-text"
          aria-invalid="true"
          autocomplete="email"
          class="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
          data-testid="email"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root-93 MuiFormHelperText-error-94"
        id="email-helper-text"
      >
        Enter your email.
      </p>
    </div>
    <div
      class="SubmitButton-root-53 SubmitButton-submitInPage-56"
    >
      <button
        class="MuiButtonBase-root-83 MuiButton-root-57 MuiButton-contained-68 MuiButton-containedPrimary-69 MuiButton-raised-71 MuiButton-raisedPrimary-72"
        data-testid="resetPasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-58"
        >
          Reset Password
        </span>
        <span
          class="MuiTouchRipple-root-86"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`ResetPasswordForm renders correctly 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-193 MuiFormControl-marginNormal-194 MuiFormControl-fullWidth-196"
    >
      <label
        class="MuiFormLabel-root-208 MuiInputLabel-root-197 MuiInputLabel-formControl-202 MuiInputLabel-animated-205"
        data-shrink="false"
        for="email"
      >
        Email
      </label>
      <div
        class="MuiInputBase-root-228 MuiInput-root-215 MuiInput-underline-219 MuiInputBase-fullWidth-237 MuiInput-fullWidth-222 MuiInputBase-formControl-229 MuiInput-formControl-216"
      >
        <input
          aria-invalid="false"
          autocomplete="email"
          class="MuiInputBase-input-238 MuiInput-input-223 MuiInputBase-inputType-241 MuiInput-inputType-226"
          data-testid="email"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="SubmitButton-root-245 SubmitButton-submitInPage-248"
    >
      <button
        class="MuiButtonBase-root-275 MuiButton-root-249 MuiButton-contained-260 MuiButton-containedPrimary-261 MuiButton-raised-263 MuiButton-raisedPrimary-264"
        data-testid="resetPasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-250"
        >
          Reset Password
        </span>
        <span
          class="MuiTouchRipple-root-278"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`ResetPasswordForm triggers resetPassword when email is provided 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-93 MuiFormControl-marginNormal-94 MuiFormControl-fullWidth-96"
    >
      <label
        class="MuiFormLabel-root-108 MuiFormLabel-error-111 MuiInputLabel-error-100 MuiInputLabel-root-97 MuiInputLabel-formControl-102 MuiInputLabel-animated-105"
        data-shrink="false"
        for="email"
      >
        Email
      </label>
      <div
        class="MuiInputBase-root-128 MuiInput-root-115 MuiInput-underline-119 MuiInputBase-error-134 MuiInput-error-120 MuiInputBase-fullWidth-137 MuiInput-fullWidth-122 MuiInputBase-formControl-129 MuiInput-formControl-116"
      >
        <input
          aria-describedby="email-helper-text"
          aria-invalid="true"
          autocomplete="email"
          class="MuiInputBase-input-138 MuiInput-input-123 MuiInputBase-inputType-141 MuiInput-inputType-126"
          data-testid="email"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root-185 MuiFormHelperText-error-186"
        id="email-helper-text"
      >
        Enter your email.
      </p>
    </div>
    <div
      class="SubmitButton-root-145 SubmitButton-submitInPage-148"
    >
      <button
        class="MuiButtonBase-root-175 MuiButton-root-149 MuiButton-contained-160 MuiButton-containedPrimary-161 MuiButton-raised-163 MuiButton-raisedPrimary-164"
        data-testid="resetPasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-150"
        >
          Reset Password
        </span>
        <span
          class="MuiTouchRipple-root-178"
        />
      </button>
    </div>
  </form>
</div>
`;
